aten/src/ATen/native/cuda/Copy.cu

```diff
@@ -200,23 +200,79 @@ void copy_to_cpu(Tensor& dst, const Tensor& src) {
   });
 }
 
+void copy_from_cpu_async_(Tensor& dst, const Tensor& src) {
+  AT_ASSERT(dst.is_contiguous());
```

Comment:
These probably shouldn't be asserts; I'm not 100% it's possible to trigger this error condition from user code, but the old TH code seemed to actually give user visible error messages in this case. `AT_ASSERT` means "We, the PyTorch devs, did something wrong, please yell at us."